src/main/drivers/pwm_output_dshot_shared.c

```diff
@@ -0,0 +1,253 @@
+        return;
+    }
+
+    /*If there is a command ready to go overwrite the value and send that instead*/
+    if (pwmDshotCommandIsProcessing()) {
+        value = pwmGetDshotCommand(index);
+#ifdef USE_DSHOT_TELEMETRY
+        // reset telemetry debug statistics every time telemetry is enabled
+        if (value == DSHOT_CMD_SIGNAL_LINE_CONTINUOUS_ERPM_TELEMETRY) {
+            dshotInvalidPacketCount = 0;
+            readDoneCount = 0;
+        }
+#endif
+        if (value) {
+            motor->requestTelemetry = true;
+        }
+    }
+
+    motor->value = value;
+
+    uint16_t packet = prepareDshotPacket(motor);
+    uint8_t bufferSize;
+
+#ifdef USE_DSHOT_DMAR
+    if (useBurstDshot) {
+        bufferSize = loadDmaBuffer(&motor->timer->dmaBurstBuffer[timerLookupChannelIndex(motor->timerHardware->channel)], 4, packet);
+        motor->timer->dmaBurstLength = bufferSize * 4;
+    } else
+#endif
+    {
+        bufferSize = loadDmaBuffer(motor->dmaBuffer, 1, packet);
+        motor->timer->timerDmaSources |= motor->timerDmaSource;
+#ifdef STM32F7
+        LL_EX_DMA_SetDataLength(motor->dmaRef, bufferSize);
+        LL_EX_DMA_EnableStream(motor->dmaRef);
+#else
+        DMA_SetCurrDataCounter(motor->dmaRef, bufferSize);
+        DMA_Cmd(motor->dmaRef, ENABLE);
+#endif
+    }
+}
+
+
+#ifdef USE_DSHOT_TELEMETRY
+
+void dshotEnableChannels(uint8_t motorCount);
+
+static uint16_t decodeDshotPacket(uint32_t buffer[])
+{
+    uint32_t value = 0;
+    for (int i = 1; i < DSHOT_TELEMETRY_INPUT_LEN; i += 2) {
+        int diff = buffer[i] - buffer[i-1];
+        value <<= 1;
+        if (diff > 0) {
+            if (diff >= 11) value |= 1;
+        } else {
+            if (diff >= -9) value |= 1;
+        }
+    }
+
+    uint32_t csum = value;
+    csum = csum ^ (csum >> 8); // xor bytes
+    csum = csum ^ (csum >> 4); // xor nibbles
+
+    if (csum & 0xf) {
+        return 0xffff;
+    }
+    return value >> 4;
+}
+
+static uint16_t decodeProshotPacket(uint32_t buffer[])
+{
+    uint32_t value = 0;
+    for (int i = 1; i < PROSHOT_TELEMETRY_INPUT_LEN; i += 2) {
+        const int proshotModulo = MOTOR_NIBBLE_LENGTH_PROSHOT;
+        int diff = ((buffer[i] + proshotModulo - buffer[i-1]) % proshotModulo) - PROSHOT_BASE_SYMBOL;
+        int nibble;
+        if (diff < 0) {
+            nibble = 0;
+        } else {
+            nibble = (diff + PROSHOT_BIT_WIDTH / 2) / PROSHOT_BIT_WIDTH;
+        }
+        value <<= 4;
+        value |= (nibble & 0xf);
+    }
+
+    uint32_t csum = value;
+    csum = csum ^ (csum >> 8); // xor bytes
+    csum = csum ^ (csum >> 4); // xor nibbles
+
+    if (csum & 0xf) {
+        return 0xffff;
+    }
+    return value >> 4;
+}
+
+#endif
+
+
+#ifdef USE_DSHOT_TELEMETRY
+
+uint16_t getDshotTelemetry(uint8_t index)
+{
+    return dmaMotors[index].dshotTelemetryValue;
+}
+
+FAST_CODE void pwmDshotSetDirectionOutput(
+    motorDmaOutput_t * const motor, bool output
+#ifndef USE_DSHOT_TELEMETRY
```

Comment:
Wait what? According to https://github.com/betaflight/betaflight/pull/7685/files#diff-3561124b26374b7c3a6f2606e6013937R190, this whole block will only be included when `USE_DSHOT_TELEMETRY` is defined, so it should be possible to eliminate this whole block.

Reply:
Good catch, Mike. Thanks.

Reply:
Ha, now I understand. :wink: